app/Http/Controllers/Admin/AccommodationController.php

```diff
@@ -249,12 +294,64 @@ public function allocate(int $id, AllocateRequest $request)
             return redirect()->back()->withErrors(['help_request_id' => __('This help request is already resolved')]);
         }
 
-        $reservedNumber = $accommodation->helpRequests->sum('guests_number');
-        if ($reservedNumber + $helpRequest->guests_number > $accommodation->max_guests) {
+        $start = Carbon::parse($request->startDate);
+        $end = Carbon::parse($request->endDate);
+
+        //Verify is AvailabilityInterval exists
+        $selectedInterval =  $accommodation->availabilityIntervals()->whereDateStrictBetween($start, $end)->first();
+        if(empty($selectedInterval)) {
```

Comment:
This needs to also take into account accommodations without limited availability.

```suggestion
        if (empty($selectedInterval) && $accommodation->availabilityIntervals()->exists()) {
```